site/app/controllers/pdf/PDFController.php

```diff
@@ -156,7 +156,7 @@ public function savePDFAnnotation(string $gradeable_id, string $target_dir) {
         $annotation_body = [
             'file_path' => $annotation_info['file_path'],
             'grader_id' => $grader_id,
-            'annotations' => json_decode($_POST['annotation_layer'], true)
+            'annotations' => isset($_POST['annotation_layer']) ? json_decode($_POST['annotation_layer'], true) : json_decode('{}')
```

Comment:
Better / quicker to just return the constant value than going through `json_decode`.
```suggestion
            'annotations' => isset($_POST['annotation_layer']) ? json_decode($_POST['annotation_layer'], true) : [];
```